pgjdbc/src/test/java/org/postgresql/test/jdbc2/BlobTest.java

```diff
@@ -290,18 +298,183 @@ void markResetStream() throws Exception {
         LargeObjectManager lom = ((PGConnection) con).getLargeObjectAPI();
 
         long oid = rs.getLong(1);
-        LargeObject blob = lom.open(oid);
+        try (LargeObject blob = lom.open(oid)) {
+          InputStream bis = blob.getInputStream();
+
+          assertEquals('<', bis.read());
+          bis.mark(4);
+          assertEquals('?', bis.read());
+          assertEquals('x', bis.read());
+          assertEquals('m', bis.read());
+          assertEquals('l', bis.read());
+          bis.reset();
+          assertEquals('?', bis.read());
+        }
+      }
+    }
+  }
+
+  @Test
+  void markResetWithInitialOffset() throws Exception {
+    assertTrue(uploadFile(TEST_FILE, NATIVE_STREAM) > 0);
+
+    try (Statement stmt = con.createStatement()) {
+      try (ResultSet rs = stmt.executeQuery("SELECT lo FROM testblob where id = '/test-file.xml'")) {
+        assertTrue(rs.next());
+
+        LargeObjectManager lom = ((PGConnection) con).getLargeObjectAPI();
+
+        long oid = rs.getLong(1);
+        try (LargeObject blob = lom.open(oid)) {
+          blob.seek(4);
+          InputStream bis = blob.getInputStream();
+
+          assertEquals('l', bis.read());
+          bis.reset();
+          assertEquals('l', bis.read());
+          assertEquals(' ', bis.read());
+          bis.mark(4);
+          assertEquals('v', bis.read());
+          assertEquals('e', bis.read());
+          bis.reset();
+          assertEquals('v', bis.read());
+        }
+      }
+    }
+  }
+
+  @Test
+  void skip() throws Exception {
+    LargeObjectManager lom = ((PGConnection) con).getLargeObjectAPI();
+    long loid = createMediumLargeObject();
+
+    try (LargeObject blob = lom.open(loid, LargeObjectManager.READ)) {
+      InputStream bis = blob.getInputStream();
+      assertEquals(0, bis.read());
+      assertEquals(1024L, bis.skip(1024));
+      assertEquals(1, bis.read());
+      assertEquals(64 * 1024L, bis.skip(64 * 1024));
+      assertEquals(65, bis.read());
+    }
+  }
+
+  @Test
+  void skipBackwards() throws Exception {
+    LargeObjectManager lom = ((PGConnection) con).getLargeObjectAPI();
+    long loid = createMediumLargeObject();
+
+    try (LargeObject blob = lom.open(loid, LargeObjectManager.READ)) {
+      InputStream bis = blob.getInputStream();
+      assertEquals(0, bis.read());
+      assertEquals(1024L, bis.skip(1024));
+      assertEquals(1, bis.read());
+      assertEquals(0L, bis.skip(-1024));
```

Comment:
Is read-skip-read needed for "skip backwards" test?
I guess it should be enough to have something like `read(); skip(-1); read()`.

Reply:
For this test file it is, because of the structure. I guess I can change it to use the xml file instead, as this test is not exercising the buffering code.